canister/src/runtime.rs

```diff
@@ -217,3 +218,24 @@ pub fn time() -> u64 {
         .unwrap()
         .as_secs()
 }
+
+fn add_cycles_burnt_to_metric(cycles_burnt: u128) {
+    with_state_mut(|s| {
+        if let Some(metric_cycles_burnt) = &mut s.metrics.cycles_burnt {
+            *metric_cycles_burnt += cycles_burnt;
+        } else {
+            s.metrics.cycles_burnt = Some(cycles_burnt);
+        }
+    });
+}
+
+#[cfg(target_arch = "wasm32")]
+pub fn cycles_burn() {
+    let cycles_burnt = ic_cdk::api::cycles_burn(ic_cdk::api::canister_balance128());
+    add_cycles_burnt_to_metric(cycles_burnt);
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+pub fn cycles_burn() {
+    add_cycles_burnt_to_metric(1_000_000);
+}
```

Comment:
`runtime.rs` is basically a wrapper around calls to the cdk. It exists so that we can provide test-friendly alternatives to SDK calls, and it shouldn't include any other logic. I'd do this:

1. Constrain this logic to only the system API call:
```
#[cfg(target_arch = "wasm32")]
pub fn cycles_burn() -> u128 {
    ic_cdk::api::cycles_burn(ic_cdk::api::canister_balance128())
}

#[cfg(not(target_arch = "wasm32"))]
pub fn cycles_burn() {
   1_000_000
}
```

The rest of the logic, like updating a metric, doesn't belong here. I'd keep that in `heartbeat.rs` as you had before.

Reply:
Done.